src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJobInd.java

```diff
@@ -0,0 +1,106 @@
+package edu.ucsb.cs156.happiercows.jobs;
+
+import java.time.LocalDateTime;
+
+import edu.ucsb.cs156.happiercows.entities.Commons;
+import edu.ucsb.cs156.happiercows.entities.Profit;
+import edu.ucsb.cs156.happiercows.entities.User;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.repositories.CommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.ProfitRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserRepository;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContext;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContextConsumer;
+import edu.ucsb.cs156.happiercows.errors.EntityNotFoundException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import java.util.Optional;
+
+@AllArgsConstructor
+public class MilkTheCowsJobInd implements JobContextConsumer {
+
+    @Getter
+    private CommonsRepository commonsRepository;
+    @Getter
+    private UserCommonsRepository userCommonsRepository;
+    @Getter
+    private UserRepository userRepository;
+    @Getter
+    private ProfitRepository profitRepository;
+    @Getter
+    private long commonsID;
+
+    public String formatDollars(double amount) {
+        return  String.format("$%.2f", amount);
+    }
+
+    @Override
+    public void accept(JobContext ctx) throws Exception {
+        ctx.log("Starting to milk the cows");
+        Optional<Commons> commonMilkedOpt = commonsRepository.findById(commonsID);
+
+        if(commonMilkedOpt.isPresent()){
+            Commons commonMilked = commonMilkedOpt.get();
+            String name = commonMilked.getName();
+            double milkPrice = commonMilked.getMilkPrice();
+            ctx.log("Milking cows for Commons: " + name + ", Milk Price: " + formatDollars(milkPrice));
+
+            Iterable<UserCommons> allUserCommons = userCommonsRepository.findByCommonsId(commonMilked.getId());
+
+            for (UserCommons userCommons : allUserCommons) {
+                milkCows(ctx, commonMilked, userCommons);
+            }
+            
+
+            ctx.log("Cows have been milked!");
+        } else {
+            ctx.log(String.format("No commons found for id %d", commonsID));
+        }
+    }
+
+    /** This method performs the function of milking the cows for a single userCommons.
+     *  It is a public method only so it can be exposed to the unit tests
+     * @param ctx the JobContext
+     * @param commons the Commons
+     * @param userCommons the UserCommons
+     *
+     */
+
+    public void milkCows(JobContext ctx, Commons commons, UserCommons userCommons) {
```

Comment:
Compare the code here:
https://github.com/ucsb-cs156-m23/proj-happycows-m23-10am-4/blob/6231df7533e393a376dbd4edc4b4a06eeeb4592a/src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java#L63

Should we really have two places in the code that do the same thing?

That creates a maintenance nightmare where future devs, when trying to update something about, for example, what happens when cows are milked has to know to change it in *both places*.  That's bad.

I realize that factoring this out is a bit inconvenient, because there are a quite a few parameters that would have to be passed.   

That is, if/when the code is refactored, you might need something like a `CowMilkingHelper` or a `CowMilkingService` with a method that takes not only the current parameters of `milkCows` but also parameters for   `userCommonsRepository` and `profitRepository`.

It could also be a method of the UserCommons which would alllow you to not have to pass the UserCommons at least.  And if you remove the ctx.log debugging output (which may not actually be needed) you could eliminate that parameter as well.

Having a method that takes so many parameters is another kind of "code smell", but that's not a good reason to create duplicate code; at least with that factored out method with the large number of parameters, it will be possible to go from there with some refactoring to make things better.